ecosystem/typescript/sdk/src/providers/indexer.ts

```diff
@@ -0,0 +1,81 @@
+import axios from "axios";
+
+import { AnyNumber } from "../bcs/types";
+import { MaybeHexString } from "../hex_string";
+import { GetAccountCurrentTokensQuery, GetTokenActivitiesQuery } from "../indexer/generated/operations";
+import { GetAccountCurrentTokens, GetTokenActivities } from "../indexer/generated/queries";
+
+interface PaginationArgs {
+  offset?: AnyNumber;
+  limit?: number;
+}
+
+type GraphqlQuery = {
+  query: string;
+  variables?: {};
+};
+/**
+ * Provides methods for retrieving data from Aptos Indexer.
+ * For more detailed Queries specification see
+ * {@link https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql}
+ */
+export class IndexerClient {
+  endpoint: string;
+
+  /**
+   * @param endpoint URL of the Aptos Indexer API endpoint.
+   */
+  constructor(endpoint: string) {
+    this.endpoint = endpoint;
+  }
+
+  /**
+   * Builds a axios client call to fetch data from Aptos Indexer.
+   *
+   * @param graphqlQuery A GraphQL query to pass in the `data` axios call.
+   */
+  private async queryIndexer(graphqlQuery: GraphqlQuery): Promise<any> {
+    try {
```

Comment:
why try / catch here?

Reply:
to catch server errors

Reply:
If there's an error, the error should be propagated up and let the Apps decide what to do. Returning the errors as result is less ideal and inconsistent with other functions.

Reply:
we are returning the errors - am i missing something?

Reply:
instead of `catch` and `return`. Let the server exception bubble up.